test/images/agnhost/go.mod

```diff
@@ -0,0 +1,96 @@
+module k8s.io/kubernetes/test/images/agnhost
+
+go 1.17
+
+replace (
+	k8s.io/api => ../../../staging/src/k8s.io/api
+	k8s.io/apiextensions-apiserver => ../../../staging/src/k8s.io/apiextensions-apiserver
+	k8s.io/apimachinery => ../../../staging/src/k8s.io/apimachinery
+	k8s.io/apiserver => ../../../staging/src/k8s.io/apiserver
+	k8s.io/cli-runtime => ../../../staging/src/k8s.io/cli-runtime
+	k8s.io/client-go => ../../../staging/src/k8s.io/client-go
+	k8s.io/cloud-provider => ../../../staging/src/k8s.io/cloud-provider
+	k8s.io/cluster-bootstrap => ../../../staging/src/k8s.io/cluster-bootstrap
+	k8s.io/code-generator => ../../../staging/src/k8s.io/code-generator
+	k8s.io/component-base => ../../../staging/src/k8s.io/component-base
+	k8s.io/component-helpers => ../../../staging/src/k8s.io/component-helpers
+	k8s.io/controller-manager => ../../../staging/src/k8s.io/controller-manager
+	k8s.io/cri-api => ../../../staging/src/k8s.io/cri-api
+	k8s.io/csi-translation-lib => ../../../staging/src/k8s.io/csi-translation-lib
+	k8s.io/kube-aggregator => ../../../staging/src/k8s.io/kube-aggregator
+	k8s.io/kube-controller-manager => ../../../staging/src/k8s.io/kube-controller-manager
+	k8s.io/kube-proxy => ../../../staging/src/k8s.io/kube-proxy
+	k8s.io/kube-scheduler => ../../../staging/src/k8s.io/kube-scheduler
+	k8s.io/kubectl => ../../../staging/src/k8s.io/kubectl
+	k8s.io/kubelet => ../../../staging/src/k8s.io/kubelet
+	k8s.io/kubernetes => ../../../
```

Comment:
what in agnhost requires k8s.io/kubernetes?

Reply:
I have to spend a bit more time on this to see if this can be simplified

Reply:
For the most part, it's self-referencing agnhost subpackages I think. Imports like this:

```
"k8s.io/kubernetes/test/images/agnhost/crd-conversion-webhook/converter"
"k8s.io/kubernetes/test/images/agnhost/net/common"
"k8s.io/kubernetes/test/images/agnhost/net/nat"
```

And so on. There are a few exceptions, like:

```
"k8s.io/kubernetes/test/images/resource-consumer/common"
```

In the previous image centralization effort, I didn't centralize the resource-consumer image as there were a few problems there. I can revisit it and centralize it into agnhost.

Another exception is:

```
admissionfuzzer "k8s.io/kubernetes/pkg/apis/admission/fuzzer"
```

Which is used in ``kubernetes\test\images\agnhost\webhook\convert_test.go``, and nowhere else. It only uses ``admissionfuzzer.Funcs`` from there, which is small, and it could be copied over in order to remove this dependency, IMO.

Keep in mind that the images that were centralized into agnhost were copied over almost as is. :)

Reply:
yeah, agnhost is part of the k8s.io/kubernetes 

https://github.com/kubernetes/kubernetes/blob/bf1c923502307687958b131561f9ec857b50d18e/test/images/agnhost/agnhost.go#L23-L49


It seems that is preferably to leave it as is, the win is not worth the effort at this moment